### ISSUE TITLE
Scheduled job constructor not called on subsequent calls in containerized environment

### ISSUE THREAD

**Issue by yalsicor** [laravel/framework#33767]
- Laravel Version: 7.23.0
- PHP Version: 7.4.4
- Database Driver & Version: psql 12.3.0

### Description:
I am running Laravel in a Heroku like PaaS environment. There are no cronjobs so I use a long running schdeuler process to trigger scheduler events. When dispatching jobs on the scheduler the Job constructor only gets called once, subsequent calls seem to use the same instance and no new data can be passed to the job.

When running the scheduler via cronjobs everything works as expected.

### Steps To Reproduce:
Long running scheduler:
```
namespace App\Console\Commands;

use Carbon\Carbon;
use Illuminate\Console\Command;

class SchedulerDaemon extends Command
{
    protected $signature = 'schedule:daemon {--sleep=60}';

    protected $description = 'Call the scheduler every minute.';

    public function handle()
    {
        while (true) {
            $this->line('<info>[' . Carbon::now()->format('Y-m-d H:i:s') . ']</info> Calling scheduler');
            $this->call('schedule:run');
            sleep($this->option('sleep'));
        }
    }
}
```

In app/Console/Kernel.php:
```
protected function schedule(Schedule $schedule)
    {
        $schedule->job(new SomeJob())->everyMinute();
    }
```

My SomeJob class:
```
namespace App\Jobs;

use Carbon\Carbon;
use Illuminate\Bus\Queueable;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Foundation\Bus\Dispatchable;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Queue\SerializesModels;
use Illuminate\Support\Facades\Log;

class SomeJob implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

    public function __construct()
    {
        Log::debug(Carbon::now()->format('[Y-m-d H:i:s]') . ' This is the Job constructor.');
    }

    public function handle()
    {
        Log::debug(Carbon::now()->format('[Y-m-d H:i:s]') . ' This is the Job handler.');
    }
}
```

The logs only show the handler Log messages. What is going on here? Is there a way to circumvent the instance "caching"?

**Comment by driesvints** [comment]
Hi there,

Thanks for reporting but it looks like this is a question which can be asked on a support channel. Please only use this issue tracker for reporting bugs with the library itself. If you have a question on how to use functionality provided by this repo you can try one of the following channels:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

However, this issue will not be locked and everyone is still free to discuss solutions to your problem!

Thanks.

**Comment by yalsicor** [comment]
Thank you @driesvints, I posted my problem [here](https://laracasts.com/discuss/channels/general-discussion/scheduled-job-constructor-not-called-on-subsequent-calls-in-containerized-environment).

After digging deeper I found, that dispatching the job manually:
```
protected function schedule(Schedule $schedule)
{
    $schedule->call(function() {
        SomeJob::dispatch();
    })->everyMinute();
}
```
Now the job constructor is fired every time. So there seems to be a problem with the `$schedule->job()` function. Wether its by design I cannot say. This seems to only be the case when using long running php processes for scheduling.

Trying to run the `schedule:run` command via job queue did not work as well.
